File: common/src/main/java/com/jon/cot/common/ui/IntentIds.java
package com.jon.cot.common.ui;

import java.util.UUID;

public class IntentIds {
    public static final String EXTRA_EDIT_PRESET_PROTOCOL = randomString();
    public static final String EXTRA_EDIT_PRESET_ALIAS = randomString();
    public static final String EXTRA_EDIT_PRESET_ADDRESS = randomString();
    public static final String EXTRA_EDIT_PRESET_PORT = randomString();
    public static final String EXTRA_EDIT_PRESET_CLIENT_BYTES = randomString();
    public static final String EXTRA_EDIT_PRESET_CLIENT_PASSWORD = randomString();
    public static final String EXTRA_EDIT_PRESET_TRUST_BYTES = randomString();
    public static final String EXTRA_EDIT_PRESET_TRUST_PASSWORD = randomString();

    private static String randomString() {
        return UUID.randomUUID().toString();
    }
}

File: common/src/main/java/com/jon/common/ui/IntentIds.kt
package com.jon.common.ui

import java.util.*

object IntentIds {
    val EXTRA_EDIT_PRESET_PROTOCOL = randomString()
    val EXTRA_EDIT_PRESET_ALIAS = randomString()
    val EXTRA_EDIT_PRESET_ADDRESS = randomString()
    val EXTRA_EDIT_PRESET_PORT = randomString()
    val EXTRA_EDIT_PRESET_CLIENT_BYTES = randomString()
    val EXTRA_EDIT_PRESET_CLIENT_PASSWORD = randomString()
    val EXTRA_EDIT_PRESET_TRUST_BYTES = randomString()
    val EXTRA_EDIT_PRESET_TRUST_PASSWORD = randomString()

    private fun randomString() = UUID.randomUUID().toString()
}
